Sources/SwiftSyntaxBuilder/DeclBuildables.swift.gyb

```diff
@@ -0,0 +1,90 @@
+%{
+  from gyb_syntax_support import *
+  from gyb_syntax_support.kinds import lowercase_first_word
+  from gyb_syntax_support.kinds import SYNTAX_BASE_KINDS
+  from gyb_syntax_support.kinds import syntax_buildable_child_type
+  NODE_MAP = create_node_map()
+  # -*- mode: Swift -*-
+  # Ignore the following admonition it applies to the resulting .swift file only
+}%
+//// Automatically Generated From DeclBuildables.swift.gyb.
+//// Do Not Edit Directly!
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+% for node in SYNTAX_NODES:
+%   if node.is_buildable():
+%     has_syntax_collection_child = False
+%     for child in node.children:
+%       child_node = NODE_MAP.get(child.syntax_kind)
+%       if child_node and child_node.is_syntax_collection():
+%         has_syntax_collection_child = True
+%       end
+%     end
+%     if has_syntax_collection_child:
+extension ${node.syntax_kind} {
+  public init(
+%       init_parameters = []
+%       for child in node.children:
+%         child_node = NODE_MAP.get(child.syntax_kind)
+%         token = SYNTAX_TOKEN_MAP.get(child.syntax_kind)
+%         if child_node and child_node.is_syntax_collection():
+%             default_value = "? = { nil }" if child.is_optional else " = { .empty }"
+%             init_parameters.append("@%sBuilder %sBuilder: () -> %s%s" % (child.syntax_kind, child.swift_name, child.syntax_kind, default_value))
+%         elif child.syntax_kind is "IdentifierToken" or token and not token.text:
```

Comment:
Nitpick: I think parenthesis around `token and not token.text` would improve readability.

------

Also, I think it would be good to have some comments on the different branches to describe what kind of transformation you are doing. Like for the first `if`: Allow initializing syntax collections with result builders, etc.

You should be able to put comments in `gyb` by prefixing the line with `% #` (`%` tells `gyb` that it’s Python line `#` makes it a Python comment).

The same (or shorter abbreviated) comments would probably also help below when you’re creating the call to `self.init`.

Reply:
I have adde comments now. 
Please resolve this if you thinks it's fine